### ISSUE TITLE
A question about using GO-CAM models (SIF)

### ISSUE THREAD

**Issue by kltm** [geneontology/helpdesk#256]
From a user trying to validate a method, gene regulatory networks inference:

I am looking at model 5716c41300000422.sif, and it has molecular_function and molecular_function_2 terms,
1) What is the difference between them? Does molecular_function_2 is some unknown molecular function that we don't know anything about and there is no GOID for it, so we assigned it to root molecular_function (GO:0003674) or does it mean something else?

![image](https://user-images.githubusercontent.com/2687900/83203666-caa26880-a0fe-11ea-8a85-b9b774d024ce.png)

**Comment by cmungall** [comment]
This export appears to essentially be an rdf-level representation. I think each edge is an edge connecting two individuals, with the edge labeled with the label of the rdf:type, with a disambiguating suffix attached when there are >1 individuals with the same rdf:type.

I think we need to take a step back and think about the use cases here, what kinds of users will consume this, how they will use it

I have heard a lot of requests from people to get tsvs from go-cams. It is clear that tsvs are a convenient simple format that many bioinformaticians like.

however, it's not clear that replicating the underlying rdf as tsvs is helpful. If people need this level of granularity then the rdf is a better choice. 

We should go out and gather more information from bioinformaticians

I have strong confidence that the representation that would maximize utility, expressivity and simplicity would be a CausalTab/Signor.

 - uses IDs
 - Is entity-centric, i.e. the base representation is a PPI, this is the level most people want
 - GO information is still in there, as additional edge properties
 - can easily be sliced and subsetted into even simpler representations

We have the conversion in one direction:

https://github.com/geneontology/signor2gocam

This also corresponds directly to what many users want to see in UIs, what we mostly have right now in the so-called "Pathway Viewer" (which is not a good term, any view of a connected CAM is a pathway view). We should probably use unified code for this.

@balhoff has a project where he is exposing Pathways in biolink which may end up looking isomorhic to a causaltab view

**Comment by kltm** [comment]
GO SIF developers, please see https://github.com/geneontology/go-site/issues/1496 instead as the development/work issue.

**Comment by nmanikgit** [comment]
Can some GO-CAM curator please verify my understanding below? 

Rahul's questions above are about the "Mouse-Atad5-B cells" GO-CAM model in the 5716c41300000422.sif file provided by GO-CAM in its Downloads page.  

1) My understanding is that molecular_function and molecular_function_2 in the sif file refers to two distinct biochemical activities. This is based on Noctua's view of the same GO-CAM model here: http://noctua.geneontology.org/editor/graph/gomodel:5716c41300000422?model_id=gomodel:5716c41300000422 . 

2) For the Bcell, Bcell_2, Bcell_3, ..., Bcell_5 in the sif file, my understanding is that they all refer to the same entity "Bcells". The different suffixes may be due to some artefact of the SIF-export process. Am I right? 

Thanks very much in advance!
Mani

**Comment by ukemi** [comment]
I think there is a misunderstanding. This model represents an uncertainty about whether the functions are the same rather than a certainty that they are different.  Due to the nature of the phenotypic analysis and in the absence of other evidence, it is not supported that the function of the gene product's causal effect on the different processes is due to the same function or a different function. It could be the same or it could be different. As additional evidence is examined, hopefully this can be resolved and the model can be adjusted. The same goes for the cells. Just looking at the model, it isn't clear from the evidence that the function that is occurring in the B cell is causally upstream of the process that occurring in the same B cell. This would require experiments to test that the effect is cell autonomous. It could be that the generic function is occurring in one B cell an is involved in a process in that cell. However, that process could in some way signal to another B cell to result in the causal effect of the process uncovered by the mutation. I would be willing to conclude that the B cell in which the isotype switching and the bond hydrolysis is occurring is the same cell because of the part_of relationship between those two processes. The same rationale would apply to the function that is part of the replication process.  Although I should go back and check why there is a part_of relationship there when the function is unknown, but at the time I must have had a reason to annotate that the function was an integral part of the process. 
The bottom line is that this model expresses uncertainty rather than certainty and it is unsafe to assume that the individuals are in fact distinct.
I hope this makes sense in that it tries to best represent the interpretation of the data without over-interpreting it.

**Comment by nmanikgit** [comment]
Thanks very much for your detailed explanation! I think I better understand the semantics of the GO-CAM models now. 

I needed to understand it to basically try to extract a list of gene->gene causal relations from the GO-CAM models (of the form "Gene A causes Gene B", both directly or indirectly through other intermediates). I understand that rdf models are the best format to extract these relations, but I haven't fully figured out what rules (such as the ones listed here for inferring new relations from existing ones: http://geneontology.org/docs/ontology-relations/) to apply to infer the gene->gene causal relations. 

Would it be possible for you to provide me any brief pointers on how one would use (i) causal relations between a gene and a process (GO term) in a GO-CAM model and (ii) annotations of genes to that process (GO term), to actually extract gene->gene causal relations? Thanks for any form of help in advance!

**Comment by balhoff** [comment]
@nmanikgit within Noctua we use a reasoner that infers indirect relations from the asserted statements. The rules for inference come from the ontologies used, e.g. GO and in particular Relation Ontology (RO) for relation inference. You can use this reasoner separately from Noctua: https://github.com/balhoff/arachne/releases/tag/v1.2.1

One workflow would be to compute inferences with arachne, save as RDF, then do a SPARQL query on the result for the kinds of patterns you are looking for. It may be that RO doesn't include all the rules you want; you can add them yourself when running arachne, or just account for that in the patterns you query.

So far we haven't had many (any?) external users run arachne; there isn't much in the way of documentation. If you want to try this, I would be happy to help you get started.

**Comment by rahulbiswas876** [comment]
@balhoff Thanks for providing reference and possible workflow. Use of Arachne for inference looks promising and interesting.
